elf-size-analyze.py

```diff
@@ -1039,6 +1069,23 @@ def print_tree(header, symbols):
         for line in lines:
             line.print()
 
+    def create_json(header, symbols):
+        tree = SymbolsTreeByPath(symbols)
+        if not args.no_merge_paths:
+            tree.merge_paths(args.fish_paths)
+        if not args.no_cumulative_size:
+            tree.accumulate_sizes()
+        if args.sort_by_name:
+            tree.sort(key=lambda symbol: symbol.name, reverse=False)
+        else:  # sort by size
+            tree.sort(key=lambda symbol: symbol.size, reverse=True)
+        if not args.no_totals:
+            tree.calculate_total_size()
+        min_size = math.inf if args.files_only else args.min_size
+        nodedict = tree._generate_node_dict(min_size=min_size)
+        with open(args.elf.replace(".elf", "") + "_" + header + "Analysis.json", "w") as jsonFile:
+            json.dump(nodedict, jsonFile) 
```

Comment:
Instead of automatically generating file name, I'd suggest outputting directly to STDOUT, so just `print(json.dumps(nodedict))`. 

This avoids potential confusion on what will be the generated file name. User can just run:
```
./elf-size-analyze.py my_file.elf -F -j > my_file.json
```
And for the case when multiple output sections are requested, there will be one JSON per line in the output, so it should be pretty easy to parse. Otherwise, users can just run the program multiple times with different arguments and redirect output to different files, e.g.
```
./elf-size-analyze.py my_file.elf -F -j > my_file_rom.json
./elf-size-analyze.py my_file.elf -R -j > my_file_ram.json
```

I think later we could also provide a command line argument to specify output file name. This could look like `--output-file somefile` and it would just redirect stdout to a file internally, e.g. something like [this](https://stackoverflow.com/a/4675744). But I think this is not very important.